src/components/app/flyouts/settings/settings.js

```diff
@@ -75,8 +62,18 @@ export class Settings extends LinkedComponent {
     this.setState({ [name]: value });
   };
 
+  onSimulationChange = ({ target }) => {
+    this.onChange({target});
```

Comment:
Since you are setting the state in this method already, it would be better to set the state for the onChange here as well instead of having multiple back to back state updates.